src/webui/www/private/scripts/dynamicTable.js

```diff
@@ -860,42 +860,43 @@ window.qBittorrent.DynamicTable ??= (() => {
             }
             else {
                 const trs = [...this.getTrs()];
+                const trMap = new Map(trs.map(tr => [tr.rowId, tr]));
 
                 for (let rowPos = 0; rowPos < rows.length; ++rowPos) {
                     const rowId = rows[rowPos].rowId;
-                    let tr_found = false;
-                    for (let j = rowPos; j < trs.length; ++j) {
-                        if (trs[j].rowId === rowId) {
-                            tr_found = true;
-                            if (rowPos === j)
-                                break;
-                            trs[j].inject(trs[rowPos], "before");
-                            const tmpTr = trs[j];
-                            trs.splice(j, 1);
-                            trs.splice(rowPos, 0, tmpTr);
-                            break;
-                        }
-                    }
-                    if (tr_found) { // row already exists in the table
-                        this.updateRow(trs[rowPos], fullUpdate);
+                    const existingTr = trMap.get(rowId);
+                    if (existingTr !== undefined) {
+                        this.updateRow(existingTr, fullUpdate);
                     }
-                    else { // else create a new row in the table
+                    else {
                         const tr = this.createRowElement(rows[rowPos]);
 
-                        // Insert
-                        if (rowPos >= trs.length) {
-                            tr.inject(this.tableBody);
-                            trs.push(tr);
-                        }
-                        else {
-                            tr.inject(trs[rowPos], "before");
-                            trs.splice(rowPos, 0, tr);
-                        }
+                        // add to end of table - we'll move into the proper order later
+                        this.tableBody.appendChild(tr);
```

Comment:
Is it able to add to a temporary list instead? So you don't need to operate on the (live?) table that much.

Reply:
This might provide another performance improvement, but I'm not sure. I think it can be handled later if this is still not performant enough.

Reply:
>This might provide another performance improvement, but I'm not sure. I think it can be handled later if this is still not performant enough.

Maybe consider add a `TODO` so we don't forget it?

Reply:
Sure - could you clarify what you mean by a temporary list? Are you talking about a virtual dom?

Reply:
I was thinking of something like `DocumentFragment`. Elements can be added to it and later add the fragment to the exact location. The `Node.insertBefore()` can add the fragment in just one call. That is the rough idea.

Also, if possible, it doesn't have to be `DocumentFragment`. It can be an `Array` and use `Element` methods plus spread syntax to add them. For example, `Element.append(...array);`